src/StreamJsonRpc.Shared/JsonRpc.cs

```diff
@@ -679,23 +680,31 @@ protected virtual Task<ReturnType> InvokeCoreAsync<ReturnType>(int? id, string t
                         var attribute = (JsonRpcMethodAttribute) method.GetCustomAttribute(typeof(JsonRpcMethodAttribute));
                         if (attribute != null)
                         {
-                            List<string> names;
-                            if (!methodToAttributeMap.TryGetValue(method.Name, out names))
+                            string value;
+                            if (methodToAttributeMap.TryGetValue(method.Name, out value))
                             {
-                                names = new List<string>();
-                                methodToAttributeMap.Add(method.Name, names);
+                                if (!string.Equals(attribute.Name, value, StringComparison.Ordinal))
+                                {
+                                    throw new RemoteTargetException(string.Format(CultureInfo.CurrentCulture, Resources.ConflictingMethodNameAttribute, method.Name, nameof(JsonRpcMethodAttribute)));
+                                }
                             }
-
-                            if (!names.Contains(attribute.Name, StringComparer.Ordinal))
+                            else
                             {
-                                names.Add(attribute.Name);
-                            }
+                                methodToAttributeMap.Add(method.Name, attribute.Name);
```

Comment:
It looks like the effect you're going for is that if I define two overloads with the same method name in C#, and only one overload bears the attribute that changes the name, both methods are impacted. That's a bit odd, but with adequate documentation on the attribute I think we can live with it.
But what if instead we invert the dictionary we're creating initially (rather than inverting it at the bottom of your method) and simply store the request method name as the key up front? Then the attribute can apply to just the one overload it was given? Of course, then you'd need to define what to do if two methods of different CLR names collide in their request method name. So IMO the way that should work is your dictionary's value is a `List<MethodInfo>` and you allow the collisions. You also add an entry to the dictionary for every single method on the type rather than only those with attributes. The dictionary then becomes the complete map of: "if you see a request method coming in with *x*, here are all the MethodInfo's to look for a compatible signature". The TargetMethod resolver can already respond to method signature ambiguity (which we can move earlier during reflection in a subsequent change to this library). 

The reason I suggest this change is I think it will lend to a more predictable API use for the consumer, and it's 'future proof' to enhancing our disambiguation logic in a few ways that I'm thinking we may want to.